cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -2230,6 +2230,15 @@ if(ARROW_TESTING)
                      1.10.0
                      USE_CONFIG
                      ${GTEST_USE_CONFIG})
+  get_target_property(gtest_cxx_standard GTest::gtest INTERFACE_COMPILE_FEATURES)
+
+  if(APPLE
+     AND (${gtest_cxx_standard} STREQUAL "cxx_std_14")
```

Comment:
GoogleTest just bumped its CMake standard [less then a year ago](https://github.com/google/googletest/blame/main/docs/quickstart-cmake.md) thus we should be fine here